tests/foreman/api/test_registration.py

```diff
@@ -126,3 +127,47 @@ def test_positive_allow_reregistration_when_dmi_uuid_changed(
     ).create()
     result = rhel_contenthost.execute(command)
     assert result.status == 0
+
+
+def test_positive_update_packages_post_registration(
+    module_target_sat,
+    module_entitlement_manifest_org,
+    module_location,
+    module_ak,
+    rhel8_contenthost,
+):
+    """Test package update on host post registration
+
+    :id: 3d0a3252-ab81-4acf-bca6-253b746f26bb
+
+    :expectedresults: Package update is successful on host post registration.
+
+    :CaseLevel: Integration
+    """
+    org = module_entitlement_manifest_org
+
+    rh_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=constants.DEFAULT_ARCHITECTURE,
+        org_id=module_entitlement_manifest_org.id,
+        product=constants.PRDS['rhel8'],
+        repo=constants.REPOS['rhst8']['name'],
+        reposet=constants.REPOSET['rhst8'],
+        releasever=None,
+    )
+    rh_repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+    rh_repo.sync()
+
+    result = rhel8_contenthost.register(
```

Comment:
register() uses hammer to register so instead use ` command = module_target_sat.api.RegistrationCommand(` as this is a api module

Reply:
Correct. I missed that. Updated to use the api method.